app/main/helpers/frameworks.py

```diff
@@ -28,8 +28,8 @@ def register_interest_in_framework(client, framework_slug):
             data={'frameworkSlug': framework_slug})
 
 
-def get_required_fields(all_fields, answers):
-    required_fields = set(all_fields)
+def get_required_fields(page_fields, answers):
```

Comment:
I think it'd be much simpler to keep this as is and apply the `& set(page_fields)` to the resulting list of required questions with a separate function. That way we can test them separately, there's no need for the `in page_fields` checks and there's no risk of missing any of the checks.